functional-tests/src/main/java/com/redhat/che/selenium/core/workspace/CheStarterWrapper.java

```diff
@@ -0,0 +1,224 @@
+      } catch (IOException e) {
+        LOG.error("Could not start process chromedriver:" + e.getMessage());
+        throw e;
+      }
+    }
+  }
+
+  public void start() throws IllegalStateException {
+    //TODO: Check if che starter is running;
+    try {
+      File cheStarterDir =
+          new File(System.getProperty("user.dir"), "target" + File.separator + "che-starter");
+
+      cloneGitDirectory(cheStarterDir);
+
+      LOG.info("Running che starter.");
+      Properties props = new Properties();
+      props.setProperty(
+          "OPENSHIFT_TOKEN_URL",
+          "https://sso." + this.osioUrlPart + "/auth/realms/fabric8/broker/openshift-v3/token");
+      props.setProperty(
+          "GITHUB_TOKEN_URL",
+          "https://auth." + this.osioUrlPart + "/api/token?for=https://github.com");
+      props.setProperty(
+          "CHE_SERVER_URL", "https://rhche." + this.osioUrlPart);
+      String pom = cheStarterDir.getAbsolutePath() + File.separator + "pom.xml";
+      EmbeddedMaven.forProject(pom)
+          .useMaven3Version("3.5.2")
+          .setGoals("spring-boot:run")
+          .setProperties(props)
+          .useAsDaemon()
+          .withWaitUntilOutputLineMathes(".*Started Application in.*", 10, TimeUnit.MINUTES)
+          .build();
+
+    } catch (GitAPIException e) {
+      throw new IllegalStateException(
+          "There was a problem with getting the git che-starter repository", e);
+    } catch (TimeoutException e) {
+      throw new IllegalStateException("The che-starter haven't started within 300 seconds.", e);
+    }
+  }
+
+  public String createWorkspace(String pathToJson, String token) throws Exception {
+    BufferedReader buffer = null;
+    try {
+      buffer = new BufferedReader(
+          new InputStreamReader(getClass().getResourceAsStream(pathToJson)));
+    } catch (Exception e) {
+      LOG.error("File with json was not found on address: " + pathToJson, e);
+      throw e;
+    }
+    String json = buffer.lines().collect(Collectors.joining("\n"));
+    String path = "/workspace";
+    MediaType JSON = MediaType.parse("application/json; charset=utf-8");
+    RequestBody body = RequestBody.create(JSON, json);
+    StringBuilder sb = new StringBuilder(this.cheStarterURL)
+        .append(path)
+        .append("?")
+        .append("masterUrl=")
+        .append(this.host)
+        .append("&")
+        .append("namespace=sth");
+    Builder requestBuilder = new Request.Builder().url(sb.toString());
+    requestBuilder.addHeader("Content-Type", "application/json");
+    requestBuilder.addHeader("Authorization", "Bearer " + token);
+    Request request = requestBuilder.post(body).build();
+    OkHttpClient client = new OkHttpClient();
+    try {
+      Response response = client.newCall(request).execute();
+      return getNameFromResponse(response);
+    } catch (IOException e) {
+      LOG.error("Workspace could not be created : "+e.getMessage(), e);
+      return null;
```

Comment:
wouldn't it be better just to throw exception when the worksapce can not be created ? Returning `null` is smth. that should be avoided as much as possible